app/webpacker/components/wca/FormBuilder/input/MarkdownEditor.js

```diff
@@ -69,17 +71,23 @@ function getOptions(disabled) {
 
   const textFormattings = ['bold', 'italic', 'heading'];
   const textStructures = ['quote', 'unordered-list', 'ordered-list', table];
-  const uploadsAndInserts = ['link', 'upload-image', map, youtube];
+  const inserts = ['link', map, youtube];
+  const uploadImageTool = editorOptions?.enableImageUpload ? ['upload-image'] : [];
   const previews = ['preview', 'side-by-side', 'fullscreen'];
   const helps = ['guide'];
   const toolbar = [
     ...textFormattings,
     '|', ...textStructures,
-    '|', ...uploadsAndInserts,
+    '|', ...inserts, ...uploadImageTool,
```

Comment:
Why don't you spread `uploadImageTool` into `inserts` directly?

Reply:
Ya I can do that too, done.